airbyte-db/lib/src/main/java/io/airbyte/db/instance/jobs/migrations/V0_35_40_001__MigrateFailureReasonEnumValues.java

```diff
@@ -0,0 +1,557 @@
+
+    ctx.update(DSL.table("attempts"))
+        .set(failureSummaryField, JSONB.valueOf(Jsons.serialize(fixedFailureSummary)))
+        .where(attemptIdField.eq(attemptId))
+        .execute();
+  }
+
+
+  /**
+   * Takes in a FailureSummary and replaces deprecated enum values with their updated versions.
+   */
+  private static AttemptFailureSummaryForMigration getFixedAttemptFailureSummary(final AttemptFailureSummaryForMigration failureSummary) {
+    final Map<FailureType, FailureType> oldFailureTypeToFixedFailureType = ImmutableMap.of(
+        FailureType.MANUAL_CANCELLATION_OLD, FailureType.MANUAL_CANCELLATION,
+        FailureType.SYSTEM_ERROR_OLD, FailureType.SYSTEM_ERROR,
+        FailureType.CONFIG_ERROR_OLD, FailureType.CONFIG_ERROR
+    );
+
+    final Map<FailureOrigin, FailureOrigin> oldFailureOriginToFixedFailureOrigin = ImmutableMap.of(
+        FailureOrigin.REPLICATION_WORKER, FailureOrigin.REPLICATION
+    );
+
+    final List<FailureReasonForMigration> fixedFailureReasons = new ArrayList<>();
+
+    failureSummary.getFailures().stream().forEach(failureReason -> {
+      final FailureType failureType = failureReason.getFailureType();
+      final FailureOrigin failureOrigin = failureReason.getFailureOrigin();
+
+      // null failureType is valid and doesn't need correction
+      if (failureType != null) {
+        if (oldFailureTypeToFixedFailureType.containsKey(failureType)) {
+          failureReason.setFailureType(oldFailureTypeToFixedFailureType.get(failureType));
+        } else if (failureType.equals(FailureType.UNKNOWN)) {
+          failureReason.setFailureType(null);
+        }
+      }
+
+      // null failureOrigin is valid and doesn't need correction
+      if (failureOrigin != null) {
+        if (oldFailureOriginToFixedFailureOrigin.containsKey(failureOrigin)) {
+          failureReason.setFailureOrigin(oldFailureOriginToFixedFailureOrigin.get(failureOrigin));
+        } else if (failureOrigin.equals(FailureOrigin.UNKNOWN)) {
+          failureReason.setFailureOrigin(null);
+        }
+      }
+
+      fixedFailureReasons.add(failureReason);
+    });
+
+    failureSummary.setFailures(fixedFailureReasons);
+    return failureSummary;
+  }
+
+
+  /**
+   * The following classes are essentially a copy of the FailureReason and AttemptFailureSummary classes at the time of this migration. They support
+   * both deprecated and new enum values and are used for record deserialization in this migration because in the future, the real FailureReason class
+   * will have those deprecated enum values removed, which would break deserialization within this migration.
+   */
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  @JsonPropertyOrder({
+      "failureOrigin",
+      "failureType",
+      "internalMessage",
+      "externalMessage",
+      "metadata",
+      "stacktrace",
+      "retryable",
+      "timestamp"
+  })
+  static
+  class FailureReasonForMigration {
```

Comment:
The formatting is a little weird here - why are `static` and `class` on different lines?

Reply:
Oh weird, maybe copy/paste and autoformatting did something wonky